torch/distributions/transforms.py

```diff
@@ -360,6 +362,33 @@ def log_abs_det_jacobian(self, x, y):
         return -(y.reciprocal() + (1 - y).reciprocal()).log()
 
 
+class TanhTransform(Transform):
+    r"""
+    Transform via the mapping :math:`y = \tanh(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.interval(-1.0, 1.0)
+    bijective = True
+    sign = +1
+
+    @staticmethod
+    def atanh(x):
+        return 0.5 * (x.log1p() - (-x).log1p())
+
+    def __eq__(self, other):
+        return isinstance(other, TanhTransform)
+
+    def _call(self, x):
+        return x.tanh()
+
+    def _inverse(self, y):
+        eps = torch.finfo(y.dtype).eps
+        return self.atanh(y.clamp(min=-1. + eps, max=1. - eps))
+
+    def log_abs_det_jacobian(self, x, y):
+        return 2. * (np.log(2.) - x - softplus(-2. * x))
```

Comment:
Shouldn't this be `math.log(2)`?

Reply:
It would be great if you could add a note explaining the reasoning here for future reference.